llvm/test/CodeGen/Mips/2008-07-06-fadd64.ll

```diff
@@ -1,8 +1,18 @@
-; RUN: llc -march=mips -mattr=single-float  < %s | FileCheck %s
+; RUN: llc -march=mips -mcpu=mips32 -mattr=single-float -O2 < %s | FileCheck %s
+; RUN: llc -march=mips -mcpu=mips32r2 -mattr=single-float -O2 < %s | FileCheck %s
 
 define double @dofloat(double %a, double %b) nounwind {
```

Comment:
The test is OK.  The bug report is about clang.

Reply:
Ohh. I see. It's interesting that it fails only for r2+.

Reply:
Because we have defined 32r2-32max support custom `ISD::BITCAST`:
```
if (Subtarget.hasMips32r2() && !Subtarget.useSoftFloat() &&
      !Subtarget.hasMips64()) {
    setOperationAction(ISD::BITCAST, MVT::i64, Custom);
  }
```